pkg/controllermanager/controller/bastion/bastion_control.go

```diff
@@ -65,19 +66,19 @@ func (c *Controller) shootAdd(ctx context.Context, obj interface{}) {
 	}
 
 	// list all bastions that reference this shoot
-	// TODO: this should be done via a field-selector
 	bastionList := operationsv1alpha1.BastionList{}
-	listOptions := client.ListOptions{Namespace: shoot.Namespace}
+	listOptions := client.ListOptions{
+		Namespace:     shoot.Namespace,
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.shootRef.name": shoot.Name}),
```

Comment:
I guess this cannot work without additional changes :
1) (for the case when the client is not cached one) it seems that gardener-apiserver does not support such field selector:
```
$ k get bastions -A --field-selector=spec.shootRef.name=foo
Error from server (BadRequest): Unable to find "operations.gardener.cloud/v1alpha1, Resource=bastions" that match label selector "", field selector "spec.shootRef.name=foo": field label not supported: spec.shootRef.name
```

2) (for the case when the client is cached one) this would require the field to be indexed

Reply:
Is there any reference that I can use to implement 2?

Reply:
You can grep for `IndexField` func usages.